microsoft-iis-mixin/alerts/alerts.libsonnet

```diff
@@ -0,0 +1,91 @@
+{
+  prometheusAlerts+:: {
+    groups+: [
+      {
+        name: 'microsoft-iis',
+        rules: [
+          {
+            alert: 'MicrosoftIISHighNumberOfRejectedAsyncIORequests',
+            expr: |||
+              sum by (job, instance, site) (increase(windows_iis_rejected_async_io_requests_total[5m])) > %(alertsWarningHighRejectedAsyncIORequests)s
```

Comment:
this approach would drop any custom labels users might put on top of default windows ones. Please consider to use sum without (x,y,z) instead.

Reply:
Do you mean something like 
` sum without(site,job,instance)(increase(windows_iis_rejected_async_io_requests_total[5m])) > %(alertsWarningHighRejectedAsyncIORequests)s
`

Reply:
@v-zhuravlev I'm not sure we want to avoid `sum by`'s here as this changes the scope of some of these alerts. I see your concern, although we haven't done anything special for custom labels in the past.

Reply:
@v-zhuravlev if we were to remove the `sum by ` for `sum` we would then get a combination of all the different attributes  for this metric. At some level we do want to sum for a singular site, since what we want is for a singular site on an instance and  specified job. I think for now we should sum by (job, instance, site) and forfeit the other labels since the alert in the spec doc says it should be by site.

Reply:
we just need to reverse the logic and use actual labels we want to drop in without(). That way we can achieve same result but keep possible custom labels. See more explanation here https://promlabs.com/blog/2022/12/11/avoid-these-6-mistakes-when-getting-started-with-prometheus/#mistake-2-aggregating-away-valuable-labels-in-alerting-expressions

Reply:
@v-zhuravlev ah, thanks for filling me in! I didn't even know about this functionality.